python/semantic_kernel/connectors/memory/redis/redis_memory_store.py

```diff
@@ -0,0 +1,530 @@
+        """
+
+        self._database = redis.Redis.from_url(connection_string)
+        if settings:
+            self.configure(settings)
+        assert self.ping(), "Redis could not establish a connection"
+
+        self._ft = self._database.ft
+        self._logger = logger or NullLogger()
+        self._vector_type = (
+            np.float32 if RedisMemoryStore.VECTOR_TYPE == "FLOAT32" else np.float64
+        )
+
+    def configure(self, settings: Dict[str, Any]) -> None:
+        """
+        Configures the Redis database connection.
+        See documentation for accepted parameters: https://redis.io/commands/config-set/
+
+        Arguments:
+            settings {Dict[str, Any]} -- Desired configuration formatted each as {parameter: value}
+
+        Example:
+            ```
+            # Create a default Redis data store
+            redis_ds = RedisMemoryStore()
+
+            # Set the host and port to be localhost:6369 and authenticate with the password "redis"
+            redis_ds.configure({"bind":"localhost", "port":6379, password="redis"})
+            ```
+
+        Exceptions:
+            Redis documentation for exceptions that can occur: https://redis.readthedocs.io/en/stable/exceptions.html
+        """
+        for param, val in settings.items():
+            try:
+                self._database.config_set(param, val)
+            except Exception as e:
+                self._logger.error(e)
+                raise e
+
+    def ping(self) -> bool:
+        """
+        Pings the Redis database connection.
+
+        Returns:
+            True if the connection is reachable, False if not
+        """
+        return self._database.ping()
+
+    async def create_collection_async(
+        self,
+        collection_name: str,
+        vector_dimension: int = 128,
+    ) -> None:
+        """
+        Creates a collection, implemented as a Redis index containing hashes
+        prefixed with "collection_name:".
+        If a collection of the name exists, it is left unchanged.
+
+        Note: vector dimensionality for a collection cannot be altered after creation.
+
+        Arguments:
+            collection_name {str} -- Name for a collection of embeddings
+            vector_dimension {int} -- Size of each vector, default to 128
+        """
+        try:
+            self._ft(collection_name).info()
+            self._logger.info(f'Collection "{collection_name}" already exists.')
+            print(f'Collection "{collection_name}" already exists.')
+        except ResponseError:
+            if vector_dimension <= 0:
+                self._logger.error("Vector dimension must be a positive integer")
+                raise Exception("Vector dimension must be a positive integer")
+
+            index_def = IndexDefinition(
+                prefix=f"{collection_name}:", index_type=IndexType.HASH
+            )
+            schema = (
```

Comment:
I'll leave the `timestamp` field as a TextField within the schema because the `datetime` module has easy conversion between a string to its class type.